lib/samson/hooks.rb

```diff
@@ -48,6 +48,13 @@ def view(name, partial)
         hooks(name) << partial
       end
 
+      def with_callback(name, block)
+        hooks(name) << block
+        yield
+      ensure
+        hooks(name).pop
+      end
```

Comment:
Something about this method rubs me the wrong way. Maybe it's having a single method that takes 2 different blocks, and it's not clear which one is the callback. Maybe just having a comment for the method explaining its purpose would clear it up.